### PR TITLE
fix(input): add v-model support

### PR DESCRIPTION
**Pull request by dine**
https://vuejs.org/v2/guide/components.html#Using-v-model-on-Components

### CODE REVIEW

File: src/components/ui-input.vue [hunk 1]
```diff
@@ -35,6 +35,14 @@ export default {
...
     },
     computed: {
+        inputListeners: function () {
+            var vm = this;
+            return Object.assign({}, this.$listeners, {
+                input: function (event) {
+                    vm.$emit('input', event.target.value);
+                },
+            });
+        },
```
Comment (bartderijk) [inline src/components/ui-input.vue:45]: Nice! Could be tweaked a little like so but that's just nitpicking
`inputListeners() {
            const vm = this;
            return Object.assign({}, this.$listeners, {
                input(event) {
                    vm.$emit('input', event.target.value);
                },
            });
        },`

Comment (TobbenTM) [reply src/components/ui-input.vue:45]: I'm a bit confused, what does this change actually achieve?
If you have a `<ui-input v-model="inputValue" />`, it'll do the exact same thing with or without this change?

Comment (TobbenTM) [reply src/components/ui-input.vue:45]: Maybe this is a very good use case for a unit test or two?

Comment (dine) [reply src/components/ui-input.vue:45]: @TobbenTM It's not the same, so I thought this change could be useful.
https://vuejs.org/v2/guide/components.html#Using-v-model-on-Components
Screenshot 2021-06-18 at 16 20 44

File: src/__tests__/ui-input.test.js [hunk 2]
```diff
@@ -38,4 +39,11 @@ describe('UiInput', () => {
         expect(wrapper.text()).toContain(SLOT_SUFIX);
         expect(input.classes()).toContain('pr-12');
     });
+
+    it('Should emit input event with value', () => {
+        vm.$emit = jest.fn();
```
Comment (TobbenTM) [inline src/__tests__/ui-input.test.js:44]: Maybe personal opinion, but I'd rather avoid mocking stuff if we don't have to.
This test can be verified by waiting a tick (`await vm.$nextTick()`), then `expect(wrapper.emitted().input[0]).toEqual([value])`.
Thoughs?

Comment (dine) [reply src/__tests__/ui-input.test.js]: No preference, so I updated the PR

**Review by TobbenTM — APPROVED** [review]
👌

**Review by TobbenTM — APPROVED** [review]

**Pull request merged**